libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -461,6 +461,13 @@ Compass::Compass(void)
     }
     _singleton = this;
     AP_Param::setup_object_defaults(this, var_info);
+
+    // Reset dev_id parameters' cached values to zero. This is here to address a potential
+    // edge case where a previously-detected compass is no longer present, but its dev_id
+    // is loaded from the parameter's stored value.
+    for (uint8_t i=0; i<COMPASS_MAX_INSTANCES; i++) {
+        _state[i].dev_id = 0;
+    }
```

Comment:
This doesn't actually do anything and has been removed recently. Reason is that it should be 0 on construction and parameters aren't loaded at this point, so it will be overwritten when parameters are loaded after this.

Reply:
fixed